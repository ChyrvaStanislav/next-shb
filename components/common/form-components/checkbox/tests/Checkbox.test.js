import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';
import Checkbox from '../index';

const onChangeFunc = jest.fn();
const setFieldTouchedFunc = jest.fn();
const fieldOnBlurFunc = jest.fn();

const Component = props => (
  <Formik>
    <Checkbox
      field={{
        value: false,
        onChange: jest.fn(),
        onBlur: fieldOnBlurFunc,
        name: 'Name',
      }}
      form={{
        touched: {},
        errors: {},
        setFieldTouched: setFieldTouchedFunc,
      }}
      id="someId"
      formId="formId"
      label="Some label"
      name="Some name"
      {...props}
    />
  </Formik>
);


describe('Checkbox renders', () => {
  it('should Checkbox renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Functions works', () => {
  it('should onChange work', () => {
    const { getByTestId } = render(<Component onChange={onChangeFunc} />);
    const checkboxElem = getByTestId('checkbox');
    checkboxElem.click();
    expect(onChangeFunc).toBeCalled();
    expect(setFieldTouchedFunc).toBeCalled();
  });
});

describe('Checkbox disable', () => {
  it('should Checkbox disable when disabled = true', () => {
    const { getByTestId } = render(<Component disabled={true} />);
    const checkboxElem = getByTestId('checkbox');
    expect(checkboxElem).toBeDisabled();
  });
});
