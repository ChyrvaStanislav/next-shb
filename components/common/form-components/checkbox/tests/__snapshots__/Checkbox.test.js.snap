// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders should Checkbox renders 1`] = `
<DocumentFragment>
  <div>
    <section
      class="row"
    >
      <label
        class="checkboxContainer"
      >
        <input
          class="checkboxHidden"
          data-testid="checkbox"
          id="formId-someId"
          name="Name"
          type="checkbox"
          value="false"
        />
        <span
          class="checkbox"
          for="formId-someId"
        />
        <span
          class="inputLabel"
        >
          Some label
        </span>
      </label>
    </section>
  </div>
</DocumentFragment>
`;
